src/index.ts

```diff
@@ -31,55 +31,54 @@ async function fetchArtifacts(): Promise<Response<ArtifactsResponse>> {
 
 async function fetchArtifactsDetails(artifact: Artifact): Promise<Response<string>> {
 
-    // TODO: DEBUG log here
     let artifactPath = '';
     if (artifact.groupId !== undefined) {
-        // TODO: DEBUG log here
         artifactPath = `v1/groups/${artifact.groupId}/artifacts/${artifact.id}`
+        Logger.debug(`Artifact has group, using resource path: ${artifactPath}`)
     } else {
-        // TODO: DEBUG log here
         artifactPath = `v1/artifacts/${artifact.id}`
+        Logger.debug(`Using resource path: ${artifactPath}`)
     }
 
+    Logger.debug(`Fetching artifact details at: ${PREFIX}/${artifactPath}`)
     return await got(artifactPath, {
         prefixUrl: PREFIX
     });
-
-    // TODO: DEBUG log here
 }
 
 function extractSubgraphUrl(artifact: Artifact) {
 
-    // TODO: DEBUG log here
+    Logger.debug(`Extracting Subgraph URL from artifact: ${artifact.id}`)
     let url = ''
 
     if (artifact.labels !== undefined) {
         for (const label of artifact.labels) {
             if (label.startsWith('xjoin-subgraph-url=')) {
-                // TODO: DEBUG log here
                 url = label.split('xjoin-subgraph-url=')[1]
             }
         }
     }
 
-    // TODO: DEBUG log here
+    Logger.debug(`Extracted URL from artifact: ${url}`)
     return url
 }
 
 async function loadSubgraphSchemas(): Promise<ServiceDefinition[]> {
-    // TODO: INFO log here
+
+    Logger.info("Fetching artifacts from Schema Registry")
     const artifactsRespose: Response<ArtifactsResponse> = await fetchArtifacts();
 
     const serviceDefinitions: ServiceDefinition[] = [];
     const artifacts: Artifact[] = artifactsRespose.body.artifacts;
 
     if (artifacts.length == 0) {
-        // TODO: INFO log here
+        Logger.warn("No artifacts found at the Schame Registry")
```

Comment:
Minor typo, I think you meant Schema Registry here.

Reply:
What a shame! 😂